### PR TITLE
Creates OfflineTransactionDispute event publishing

### PR DESCRIPTION
**Pull request by clarissalimab**
Closes [tix #2974](https://github.com/CommitChange/tix/issues/2974).

### CODE REVIEW

**Comment by clarissalimab** [comment]
@wwahammy this is ready for review! (I can't request your review, the button is disabled)

**Comment by wwahammy** [comment]
I'll clean it up now since it's not a huge issue but in the future, make sure to only include changes that relate to the current PR, not other PRs for a pull request. In this case, you had all the offline transaction refund changes in the repo. The problem is that it makes it really confusing to separate out what changed specifically in this changes versus other changes.

In general, branch from the upstream branch. The only exception is when a PR literally depends on another PR. In that case, you may include the stuff form the other PR but make sure to mention it. Does that makes sense @clarissalimab?

**Comment by clarissalimab** [comment]
@wwahammy sure it does! I just included the changes from OfflineTransactionRefund in this case because I knew it was going to be merged before and the diff would be updated.

**Comment by wwahammy** [comment]
@clarissalimab: could you rebase on the latest upstream and clean up the history? When you do, please let me know it's ready for review

File: db/schema.rb [hunk 1]
```diff
@@ -606,19 +606,18 @@
...
-  create_table "offline_transaction_disputes", force: :cascade do |t|
+    t.index ["payment_id"], name: "index_offline_transaction_disputes_on_payment_id"
+  end
+
+  create_table "offline_transaction_refunds", id: :string, force: :cascade do |t|
     t.bigint "payment_id"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
-    t.index ["payment_id"], name: "index_offline_transaction_disputes_on_payment_id"
->>>>>>> 29887023... Initial offline_transaction_dispute support
```
Comment (clarissalimab) [inline db/schema.rb]: Some rebase was wrong here, I think it would be better to fix it on your branch!

Comment (wwahammy) [reply db/schema.rb]: Dang it, that's on me. I just force pushed.

**Comment by clarissalimab** [comment]
@wwahammy I just updated it!

**Review by wwahammy — APPROVED** [review]
It looks good!

**Pull request merged**